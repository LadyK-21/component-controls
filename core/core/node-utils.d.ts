export * from './dist/node-utils.d';
