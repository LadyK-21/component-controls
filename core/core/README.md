# Table of contents

-   [Overview](#overview)
-   [Installation](#installation)
-   [API](#api)

# Overview

Type definitions of the component-controls specification and accompanying utility functions.
Includes definitions for:

-   [Story](https://github.com/ccontrols/component-controls/tree/master/core/core#story)
-   [Stories](https://github.com/ccontrols/component-controls/tree/master/core/core#stories)
-   [ControlTypes](https://github.com/ccontrols/component-controls/tree/master/core/core#controltypes)
-   [ComponentControl](https://github.com/ccontrols/component-controls/tree/master/core/core#componentcontrol)
-   [PropTypes](https://github.com/ccontrols/component-controls/tree/master/core/core#proptypes)
-   and more...

# Installation

This package is usually installed as part of the @component-controls package, but you can also install it standalone:

```bash
$ npm install @component-controls/core --save-dev
```

# API

<tsdoc-typescript files="@types/webpack/index.d.ts" entry="./src/document.ts,./src/controls.ts,./src/components.ts,./src/propsInfo.ts,./src/utility.ts,./src/configuration.ts,./src/build.ts"/>

<!-- START-TSDOC-TYPESCRIPT -->

<!-- END-TSDOC-TYPESCRIPT -->
