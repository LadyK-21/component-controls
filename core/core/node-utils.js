module.exports = require('./dist/node-utils');
