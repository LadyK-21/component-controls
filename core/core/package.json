{
  "name": "@component-controls/core",
  "version": "4.0.2",
  "description": "Component controls core types and utility functions",
  "main": "dist/index.js",
  "module": "dist/index.esm.js",
  "typings": "dist/index.d.ts",
  "files": [
    "dist/",
    "package.json",
    "README.md",
    "node-utils.js",
    "node-utils.d.ts"
  ],
  "scripts": {
    "build": "yarn cross-env NODE_ENV=production rollup -c",
    "dev": "yarn cross-env NODE_ENV=development rollup -cw",
    "docs": "api-readme",
    "fix": "yarn lint --fix",
    "lint": "yarn eslint . --ext mdx,ts,tsx",
    "prepare": "yarn build",
    "test": "yarn jest"
  },
  "homepage": "https://github.com/ccontrols/component-controls",
  "bugs": {
    "url": "https://github.com/ccontrols/component-controls/issues"
  },
  "repository": {
    "type": "git",
    "url": "https://github.com/ccontrols/component-controls.git",
    "directory": "core/core"
  },
  "license": "MIT",
  "dependencies": {
    "@babel/types": "^7.12.6",
    "@component-controls/follow-imports": "^4.0.2",
    "@component-controls/logger": "^4.0.2",
    "@structured-types/api": "^3.46.7",
    "@structured-types/typescript-config": "^3.40.5",
    "@storybook/csf": "^0.0.1",
    "deepmerge": "^4.2.2",
    "escape-html": "^1.0.3",
    "esm": "^3.2.25",
    "faker": "^4.1.0",
    "path": "^0.12.7",
    "tmp": "^0.2.1",
    "typescript": "^4.2.4"
  },
  "devDependencies": {
    "@types/faker": "^4.1.9",
    "@types/jest": "^26.0.10",
    "@types/react": "^16.9.34",
    "cross-env": "^5.2.1",
    "eslint": "^6.5.1",
    "jest": "^26.4.2",
    "react": "^17.0.1",
    "webpack": "^5.24.3"
  },
  "peerDependencies": {
    "react": "^16.8.0 || ^17.0.0"
  },
  "publishConfig": {
    "access": "public"
  },
  "authors": [
    "Atanas Stoyanov"
  ],
  "jest": {
    "preset": "ts-jest",
    "globals": {
      "ts-jest": {
        "isolatedModules": true
      }
    },
    "snapshotSerializers": [
      "@emotion/jest/serializer"
    ]
  },
  "gitHead": "e30d62f101e104711a16261fce04785d58cac1eb"
}
