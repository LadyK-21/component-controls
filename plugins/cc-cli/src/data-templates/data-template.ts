import dot from 'dot';
import {
  getStoryControls,
  randomizeData,
  fixControlTypes,
} from '@component-controls/core';
import { prettify } from '@component-controls/instrument';
import { StoryTemplateOptions } from '../utils';
import { getStore } from '../store';
import { getTemplate } from '../resolve-template';

/**
 * create a data-driven testing template with random values
 * if some values already exist, will re-use them
 * @param options - templating options
 * @param existing - optional existing values
 * @returns the file content (to be saved) and the list of the document stories that have values
 */
export const createDataTemplate = async (
  options: StoryTemplateOptions,
  existing?: Record<string, any>,
): Promise<{ content: string; data: Record<string, any> } | undefined> => {
  const {
    name,
    bundle,
    storyPath = '',
    output,
    format = 'ts',
    data: numValues = 0,
  } = options;
  if (numValues <= 0) {
    return undefined;
  }
  const parsed = await getStore({ bundle, name, storyPath });
  if (!parsed) {
    return undefined;
  }
  const { doc, stories, components } = parsed;
  const data: Record<string, any> = {};
  Object.keys(stories).forEach(storyId => {
    const story = stories[storyId];

    const storyControls = getStoryControls(story, doc, components);
    if (storyControls) {
      const controls = fixControlTypes(storyControls);
      const values: Record<string, any> = existing?.[storyId] || {};
      for (let i = Object.keys(values).length; i < numValues; i += 1) {
        const rnd = randomizeData(controls);
        if (Object.keys(rnd).length) {
          values[i.toString()] = rnd;
        }
      }
      if (Object.keys(values).length) {
        data[storyId] = values;
      }
    }
  });
  if (!Object.keys(data).length) {
    return undefined;
  }
  return {
    content: prettify(
      dot.template(getTemplate(`data-templates/data`, format))({
        data: JSON.stringify(data, null, 2),
      }),
      {},
      output,
    ),
    data,
  };
};
