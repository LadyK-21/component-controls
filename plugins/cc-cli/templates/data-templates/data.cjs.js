exports.{{=it.story}} =
  {{=it.values }}
;
