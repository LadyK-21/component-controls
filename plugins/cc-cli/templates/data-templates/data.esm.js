export const {{=it.story}} =
  {{=it.values }}
;
