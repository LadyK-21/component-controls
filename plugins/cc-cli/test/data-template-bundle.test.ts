import path from 'path';
import { randomizeSeed } from '@component-controls/core';
import { createDataTemplate } from '../src/data-templates/data-template';

describe('data-template-bundle', () => {
  randomizeSeed(11223344);
  it('import', async () => {
    const template = await createDataTemplate({
      data: 5,
      bundle: path.resolve(__dirname, 'bundle/component-controls.js'),
      name: 'VariantButton',
    });
    expect(template).toMatchSnapshot();
  }, 50000);
});
