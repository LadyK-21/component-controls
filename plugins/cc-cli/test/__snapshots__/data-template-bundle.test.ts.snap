// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`data-template-bundle import 1`] = `
Object {
  "content": "export default {
  \\"variantbutton--overview\\": {
    \\"1\\": {
      text: \\"Gisselle Mohr\\",
      variant: \\"success\\",
      fontSize: 19,
      icon: \\"left-arrow\\",
      iconSide: \\"right\\",
      padding: \\"medium\\",
    },
    \\"2\\": {
      text: \\"Bart Bechtelar\\",
      variant: \\"error\\",
      fontSize: 17,
      icon: \\"none\\",
      iconSide: \\"right\\",
      iconSize: \\"medium\\",
      padding: \\"small\\",
    },
    \\"3\\": {
      text: \\"Lula Rippin\\",
      variant: \\"primary\\",
      fontSize: 13,
      icon: \\"right-arrow\\",
      iconSide: \\"left\\",
      iconSize: \\"large\\",
      padding: \\"small\\",
    },
    \\"4\\": {
      text: \\"Kelvin Feeney\\",
      variant: \\"accent\\",
      fontSize: 24,
      icon: \\"search\\",
      padding: \\"small\\",
    },
    \\"5\\": {
      text: \\"Leslie Mohr\\",
      variant: \\"disabled\\",
      fontSize: 13,
      icon: \\"none\\",
      iconSize: \\"small\\",
      padding: \\"small\\",
    },
  },
};
",
  "data": Object {
    "variantbutton--overview": Object {
      "1": Object {
        "fontSize": 19,
        "icon": "left-arrow",
        "iconSide": "right",
        "iconSize": undefined,
        "padding": "medium",
        "text": "Gisselle Mohr",
        "variant": "success",
      },
      "2": Object {
        "fontSize": 17,
        "icon": "none",
        "iconSide": "right",
        "iconSize": "medium",
        "padding": "small",
        "text": "Bart Bechtelar",
        "variant": "error",
      },
      "3": Object {
        "fontSize": 13,
        "icon": "right-arrow",
        "iconSide": "left",
        "iconSize": "large",
        "padding": "small",
        "text": "Lula Rippin",
        "variant": "primary",
      },
      "4": Object {
        "fontSize": 24,
        "icon": "search",
        "iconSide": undefined,
        "iconSize": undefined,
        "padding": "small",
        "text": "Kelvin Feeney",
        "variant": "accent",
      },
      "5": Object {
        "fontSize": 13,
        "icon": "none",
        "iconSide": undefined,
        "iconSize": "small",
        "padding": "small",
        "text": "Leslie Mohr",
        "variant": "disabled",
      },
    },
  },
}
`;
